doc/ApiDocGeneration/Generate-Api-Docs.ps1

```diff
@@ -0,0 +1,305 @@
+    "servermanagement"="Server Management"; "servicebus"="Service Bus";
+    "servicefabric"="Service Fabric"; "signalr"="signalR";
+    "sqlmanagement"="SQL Management"; "sqlvirtualmachine"="SQL Virtual Machine";
+    "storage"="Storage"; "storagecache"="Storage Cache";
+    "storagesync"="Storage Sync"; "storsimple"="Stor Simple";
+    "storsimple8000series"="Stor Simple 8000 series"; "streamanalytics"="Stream Analytics";
+    "subscription"="Subscription";
+    "trafficmanager"="Traffic Manager"; "websites"="Websites";
+}
+
+Function Get-DocFx
+{
+    [CmdletBinding()]
+    Param ($DownloadDirectory)
+
+    try 
+    {
+        Write-Verbose "Downloading DocFx..."
+        Invoke-WebRequest -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+        -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "$($DownloadDirectory)/docfx/"
+    }
+    catch 
+    {
+        Write-Verbose "Failed to Download DocFx"
+        Write-Error "$_"
+        Exit 1
+    }
+}
+
+Function Get-DocTools
+{
+    [CmdletBinding()]
+    Param (
+        $DownloadDirectory,
+        $RepoRoot
+    )
+
+    try 
+    {
+        Write-Verbose "Download and Extract mdoc to Build.BinariesDirectory/mdoc"
+        Invoke-WebRequest -Uri "https://github.com/mono/api-doc-tools/releases/download/mdoc-5.7.4.9/mdoc-5.7.4.9.zip" `
+        -OutFile "mdoc.zip" | Wait-Process; Expand-Archive -Path "mdoc.zip" -DestinationPath "$($DownloadDirectory)/mdoc/"
+
+        Write-Verbose "Restore doc/ApiDocGeneration/assets/docgen.csproj, to get ECMA2Yml and popimport"
+        dotnet restore "$($RepoRoot)/doc/ApiDocGeneration/assets/docgen.csproj" /p:BuildBinariesDirectory=$DownloadDirectory
+    }
+    catch 
+    {
+        Write-Verbose "Failed to Download Mdoc"
+        Write-Error "$_"
+        Exit 1
+    }
+
+    Get-DocFx -DownloadDirectory $DownloadDirectory
+}
+
+
+
+Function Build-Docs
+{
+    [CmdletBinding()]
+    Param (
+        $ArtifactName,
+        $ServiceDirectory,
+        $ArtifactsSafeName,
+        $ArtifactsDirectoryName,
+        $LibType,
+        $RepoRoot,
+        $BinDirectory
+    )
+
+    # Include if runing Build-Docs locally
+    # Get-DocTools -DownloadDirectory $BinDirectory -RepoRoot $RepoRoot
+
+    Write-Verbose "Create variables for identifying package location and package safe names"
+    $PackageLocation = "$($ServiceDirectory)/$($ArtifactName)"
+    Write-Verbose "Package Location $($PackageLocation)"
+    $SafeName = $ArtifactsSafeName
+
+    if ($ServiceDirectory -eq '*') {
+        $PackageLocation = "core/$($ArtifactName)"
+    }
+
+    if ($ServiceDirectory -eq 'cognitiveservices') {
+        $PackageLocation = "cognitiveservices/$($ArtifactsDirectoryName)"
+        $SafeName = $ArtifactsDirectoryName
+    }
+
+    if ($LibType -eq 'Management') {
+        $PackageLocation = "$($ServiceDirectory)/$($ArtifactName)"
+        $SafeName = $ArtifactsSafeName
```

Comment:
I assume you only need the second line below and that light seems to assume that the word "Management" is part of any library with libtype management. If that is the case why even have the libtype and just do the check for Management name?

Reply:
I used Libtype as a general way to distinguish management builds.